### ISSUE TITLE
Could code work segmenting building footprints from aerials?

### ISSUE THREAD

**Issue by rfairhurst** [DrSleep/light-weight-refinenet#45]
My real interest in semantic segmentation using RefineNet is to essentially replicate the workflow used by [Microsoft to extract building footprints from aerials](https://github.com/Microsoft/USBuildingFootprints).  Is the code you have posted suitable to be adapted to extract building footprints from aerials?  I know that I would have to prepare aerial image and mask chips for the training and validation data that fit the code's input requirements, but are there other changes that would have to be made to this code or its inputs to make this work for the task of segmenting building footprints?

I realize this code does not account for several aspects of the problem I am trying to solve, such as the handling of the global coordinates of the aerial tiles, but I am hoping that this code could provide a starting point for training a RefineNet enhanced building footprint semantic segmentation model.  If not, are you aware of any other sources of code that would be better suited to that task that you could share?

**Comment by DrSleep** [comment]
The code provided can be used with other datasets -- the guidance is given in README. I am not aware of your task-specific networks.

**Comment by rfairhurst** [comment]
I have tried the code with building footprints as the only class and that does not appear to work.  It seems to overfit the data and gets a validation score of 1.00 after two epochs.  I could have trained and validated with 4K photos each if the code worked for a single class.  Is it possible to train and validate based on a single class or does this code require at least two classes to work?

I am now trying to categorize a set of the training and validation photos to have 6 classes (structures, asphalt, dirt/barren, vegetation, concrete and pools/spas).  It is taking a long time to do each photo, so I will be training and validating with a very small set of photos (maybe 50 each).  I am sure that is far from ideal, but I will try it and see if it produces any usable results.
